core/geometry/src/curve/internalContexts/EllipticalArcApproximationContext.ts

```diff
@@ -0,0 +1,1092 @@
+      return { quadrant: 2, angle0: Angle.piOver2Radians, angle1: Angle.piRadians };
+    if (AngleSweep.isRadiansInStartEnd(radians0, Angle.piRadians, Angle.pi3Over2Radians)
+      && AngleSweep.isRadiansInStartEnd(radians1, Angle.piRadians, Angle.pi3Over2Radians))
+      return { quadrant: 3, angle0: Angle.piRadians, angle1: Angle.pi3Over2Radians };
+    if (AngleSweep.isRadiansInStartEnd(radians0, Angle.pi3Over2Radians, Angle.pi2Radians)
+      && AngleSweep.isRadiansInStartEnd(radians1, Angle.pi3Over2Radians, Angle.pi2Radians))
+      return { quadrant: 4, angle0: Angle.pi3Over2Radians, angle1: Angle.pi2Radians };
+    return undefined;
+  }
+};
+
+/**
+ * Interface implemented by sampler classes.
+ * * Implementation constructors are assumed to supply the sampler with the ellipse and relevant parameters.
+ * @internal
+ */
+interface EllipticalArcSampler {
+  /**
+   * Return samples interior to the first quadrant of the (full) ellipse.
+   * * Samples are returned as an unordered array of radian angles in the open interval (0, pi/2).
+   * @param result optional preallocated array to populate and return
+   * @return array of radian angles
+   */
+  computeRadiansStrictlyInsideQuadrant1(result?: number[]): number[];
+};
+
+/**
+ * Implementation for method `EllipticalArcSampleMethod.UniformParameter`
+ * @internal
+ */
+class UniformParameterSampler implements EllipticalArcSampler {
+  private _context: EllipticalArcApproximationContext;
+  private _options: EllipticalArcApproximationOptions;
+  private constructor(c: EllipticalArcApproximationContext, o: EllipticalArcApproximationOptions) {
+    this._context = c;
+    this._options = o;
+  }
+  public static create(
+    context: EllipticalArcApproximationContext, options: EllipticalArcApproximationOptions,
+  ): UniformParameterSampler {
+    return new UniformParameterSampler(context, options);
+  }
+  public computeRadiansStrictlyInsideQuadrant1(result?: number[]): number[] {
+    if (!result)
+      result = [];
+    if (this._context.isValidArc) {
+      const aDelta = Angle.piOver2Radians / (this._options.numSamplesInQuadrant - 1);
+      for (let i = 1; i < this._options.numSamplesInQuadrant - 1; ++i)
+        result.push(i * aDelta);
+    }
+    return result;
+  }
+};
+
+/**
+ * Implementation for method `EllipticalArcSampleMethod.NonUniformCurvature`
+ * @internal
+ */
+class NonUniformCurvatureSampler implements EllipticalArcSampler {
+  protected _context: EllipticalArcApproximationContext;
+  protected _options: EllipticalArcApproximationOptions;
+  private _xMag2: number;
+  private _yMag2: number;
+  private _curvatureRange: Range1d;
+  protected constructor(c: EllipticalArcApproximationContext, o: EllipticalArcApproximationOptions) {
+    this._context = c;
+    this._options = o;
+    this._xMag2 = c.arc.matrixRef.columnXMagnitudeSquared();
+    this._yMag2 = c.arc.matrixRef.columnYMagnitudeSquared();
+    // extreme curvatures occur at the axis points
```

Comment:
Extreme curvatures occur at the axis points only if major/minor axes are parallel to xy axes, right? Since, that's not always the case, I assume before getting to this line, we already moved the ellipse into xy-plane and rotated it to make it parallel with xy axes, right?

Reply:
Yes, "axis points" is vague, and is true if ellipse axes have been squared. (The ellipse doesn't have to be aligned to global xy axes, however.)  The context ctor stores the squared axis formulation of the input arc, so all logic that uses the context's arc can assume squared axes.